app/agent/utils.ts

```diff
@@ -111,10 +112,32 @@ export const withSsh = async <T>(
     15,
     500,
   );
+
+  const context = getActivityContext();
+  let finish: () => void = () => {};
+  const finished = new Promise<void>((resolve) => {
+    finish = resolve;
+  });
+
+  if (context != null) {
+    Promise.race([context.cancelled, finished]).catch(() => {
+      try {
+        ssh.connection?.destroy();
```

Comment:
Can you be sure that ssh exists when this callback is called? Aka ssh.dispose is called first.

Reply:
I can be sure that the ssh variable is valid, because dispose doesn't break it, but I can't be sure that the connection is still established. If it's not, then connection is null, and the `?` operator handles this case.